File: Java/6 CODE ERROR/4 Private Array-Typed Field Returned From A Public Method/java/MOPAS_1_private_array_field_from_public_method.java
/**
 *  MOPAS Private Array-Typed Field Returned From A Public Method Example 1
 */
import java.io.IOException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@WebServlet(urlPatterns = {"/MOPAS_1_private_array_field_from_public_method"})
public class MOPAS_1_private_array_field_from_public_method extends HttpServlet{
    private String[] colors;

    @Override
    public void doGet(HttpServletRequest request, HttpServletResponse response)
            throws IOException, ServletException {

        String[] publicColors = getColors();
    }

    public String[] getColors() {
        return colors;
    }
}

File: PHP/6 CODE ERROR/4 Private Array-Typed Field Returned From A Public Method/php/MOPAS_1_private_array_field_from_public_method.php
<?php
/**
 *  MOPAS Private Array-Typed Field Returned From A Public Method Example 1
 */
class Test {
	private $colors = array('red', 'green', 'blue');
	
	public function &GetColorsRef(){
		return $this->colors;
	}
	
	public function GetColorsVal(){
		return $this->colors;
	}
}

$test = new Test;
$colors = & $test->GetColorsRef();
$colors[] = 'black';

foreach ($test->GetColorsVal() as $color) {
	echo $color . ', ';
}
?>